sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/ClientSideRequestStatistics.java

```diff
@@ -89,14 +103,48 @@ void recordResponse(RxDocumentServiceRequest request, StoreResult storeResult) {
             }
 
             if (storeResponseStatistics.requestOperationType == OperationType.Head ||
-                    storeResponseStatistics.requestOperationType == OperationType.HeadFeed) {
+                storeResponseStatistics.requestOperationType == OperationType.HeadFeed) {
                 this.supplementalResponseStatisticsList.add(storeResponseStatistics);
             } else {
                 this.responseStatisticsList.add(storeResponseStatistics);
             }
         }
     }
 
+    void recordGatewayResponse(RxDocumentServiceRequest rxDocumentServiceRequest, StoreResponse storeResponse, CosmosClientException exception) {
+        ZonedDateTime responseTime = ZonedDateTime.now(ZoneOffset.UTC);
+        connectionMode = ConnectionMode.GATEWAY;
+        synchronized (this) {
+            if (responseTime.isAfter(this.requestEndTime)) {
+                this.requestEndTime = responseTime;
+            }
+            this.gatewayStatistic = new GatewayStatistic();
+            this.gatewayStatistic.operationType = rxDocumentServiceRequest.getOperationType();
+            if (storeResponse != null) {
+                this.gatewayStatistic.statusCode = storeResponse.getStatus();
+                this.gatewayStatistic.subStatusCode = DirectBridgeInternal.getSubStatusCode(storeResponse);
+
+                ISessionToken sessionToken = null;
+                String headerValue;
+                if ((headerValue = storeResponse.getHeaderValue(HttpConstants.HttpHeaders.SESSION_TOKEN)) != null) {
+                    sessionToken = SessionTokenHelper.parse(headerValue);
```

Comment:
We are encoding SessionToken to a string and then decode it back here to SessionToken object. and then at the end we encode it back to string in `ClientStatistics.toString()` 

can we avoid parsing SessionToken and just rely on its string value?

Reply:
done